bastion/src/supervisor.rs

```diff
@@ -1689,12 +1665,108 @@ impl Supervised {
     }
 }
 
+impl RestartStrategy {
+    /// Creates a new instance of RestartStrategy.
+    ///
+    /// # Arguments
+    ///
+    /// * `max_retries` - Defines an amout of attempts to restore an actor
+    /// to the initial state. Set `None` to have unlimited amout of attemps.
+    ///
+    /// * `strategy` - The strategy to use:
+    ///     - [`ActorRestartStrategy::Immediate`] would restart the
+    ///         failed actor as soon as possible.
+    ///     - [`ActorRestartStrategy::LinearBackOff`] would restart the
+    ///         failed actor with the delay increasing linearly.
+    ///     - [`ActorRestartStrategy::ExponentialBackOff`] would restart the
+    ///         failed actor with the delay, multiplied by given coefficient.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use std::time::Duration;
+    /// # use bastion::prelude::*;
+    /// #
+    /// # fn main() {
+    /// #     use std::time::Duration;
+    /// #     let actor_restart_strategy = ActorRestartStrategy::LinearBackOff {
+    /// #         timeout: Duration::from_secs(5)
+    /// #     };
+    /// #     let restart_strategy = RestartStrategy::default()
+    /// #        .with_actor_restart_strategy(actor_restart_strategy);
+    /// # }
+    /// ```
+    ///
+    /// [`ActorRestartStrategy::Instantly`]: supervisor/enum.ActorRestartStrategy.html#variant.Instantly
+    /// [`ActorRestartStrategy::LinearBackOff`]: supervisor/enum.ActorRestartStrategy.html#variant.LinearBackOff
+    /// [`ActorRestartStrategy::ExponentialBackOff`]: supervisor/enum.ActorRestartStrategy.html#variant.ExponentialBackOff
+    pub fn new(max_restarts: Option<usize>, strategy: ActorRestartStrategy) -> Self {
+        RestartStrategy {
+            max_restarts,
+            strategy,
+        }
+    }
+
+    /// Returns the acceptable count of retries for the failed actor.
+    /// The `None` value means the amount of attempts is unlimited.
+    pub fn max_restarts(&self) -> Option<usize> {
+        self.max_restarts
+    }
+
+    /// Returns the current strategy for restarting failed actors.
+    pub fn strategy(&self) -> ActorRestartStrategy {
+        self.strategy.clone()
+    }
+
+    /// Sets the limit of attempts for restoring failed actors.
+    pub fn with_max_restarts(mut self, max_restarts: Option<usize>) -> Self {
```

Comment:
I have surface API improvement:
* Let's make this isize
* isize = !0 means that unlimited restarts.
* isize > 0 means that use given restart count.
* isize = 0 means that don't restart.

What do you think @o0Ignition0o and @Relrin ?

Reply:
Perhaps it worth to reorganize it into enum (for better check in match expression), like this:
```
pub enum MaxRestartsPolicy {
    Unlimited,
    Limit(usize)
    DontRestart,
}
```
It looks better and easier to understand intentions of other devs from code.

Reply:
An enum sounds great
Maybe this naming would convey the meaning more accurately:
```rust
pub enum RestartPolicy {
    Always,
    Never,
    Tries(usize)
}

pub fn with_restarts(mut self, restarts: RestartPolicy) -> Self {}
```
What do you think ?

Reply:
Sounds good to me. I will refactor code with this enum